rp235x-hal-examples/README.md

```diff
@@ -56,44 +56,128 @@ https://github.com/rp-rs/rp-hal-boards/ for more details.
+RP2350 has two possible targets: `thumbv8m.main-none-eabihf` for the Arm mode, and
+`riscv32imac-unknown-none-elf` for the RISC-V mode.
 
 ```console
 $ rustup target add thumbv8m.main-none-eabihf
 $ rustup target add riscv32imac-unknown-none-elf
 ```
 
-To build all the examples for Arm mode, run:
+**Note: all examples assume the current directory is `<repo root>/rp235x-hal-examples`.**
+```
+cd rp235x-hal-examples
+```
+
+The most basic method is to use `cargo build` with the `--bin` flag to specify the example you want to
+build. For example, to build the `blinky` example:
 
 ```console
-$ cargo build --target=thumbv8m.main-none-eabihf
-   Compiling rp235x-hal-examples v0.1.0 (/home/user/rp-hal/rp235x-hal-examples)
-    Finished `dev` profile [unoptimized + debuginfo] target(s) in 4.53s
 $ cargo build --bin blinky
-    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
+   Compiling proc-macro2 v1.0.89
+   Compiling unicode-ident v1.0.13
+   Compiling syn v1.0.109
+...
+   Compiling pio-parser v0.2.2
+   Compiling rp235x-hal v0.2.0
+   Compiling pio-proc v0.2.2
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 14.97s
+```
+
+This builds the default target, which is Arm mode and the ELF file
+is located at `./target/thumbv8m.main-none-eabihf/debug/blinky`:
+
+```console
 $ file ./target/thumbv8m.main-none-eabihf/debug/blinky
 ./target/thumbv8m.main-none-eabihf/debug/blinky: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, with debug_info, not stripped
 ```
 
-You can also 'run' an example, which thanks to our supplied
-[`.cargo/config.toml`](./.cargo/config.toml) will invoke Raspberry Pi's
-`picotool` to copy it to an RP235x in USB Bootloader mode. You should install
-that if you don't have it already, from
-<https://github.com/raspberrypi/pico-sdk-tools/releases>.
+If you want to build the RISC-V mode you specify the target directly to override the default:
+
+```console
+$ cargo build --target=riscv32imac-unknown-none-elf --bin blinky
+   Compiling nb v1.1.0
+   Compiling byteorder v1.5.0
+   Compiling stable_deref_trait v1.2.0
+..
+   Compiling futures v0.3.31
+   Compiling frunk v0.4.3
+   Compiling rp235x-hal v0.2.0
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 6.23s```
+```
+
+And we see that the RISC-V mode ELF file is now present at `./target/riscv32imac-unknown-none-elf/debug/blinky`:
+
+```console
+$ file ./target/riscv32imac-unknown-none-elf/debug/blinky
+./target/riscv32imac-unknown-none-elf/debug/blinky: ELF 32-bit LSB executable, UCB RISC-V, RVC, soft-float ABI, version 1 (SYSV), statically linked, with debug_info, not stripped
+```
+
+You can also specify the Arm mode target directly by using
+`--target thumbv8m.main-none-eabihf` instead of `--target riscv32imac-unknown-none-elf`.
+
+To build, flash and start the application on the RP235x
+you use `cargo run` with one of the following commands:
```

Comment:
```suggestion
you use `cargo run` with one of the following commands. Note: be sure the RP235x is in BOOTSEL mode before using the `run` command because we use `picotool` to flash and run the binary:
```